aicsimageio/readers/czi_reader.py

```diff
@@ -0,0 +1,740 @@
+                self._scenes = tuple(scene_names)
+
+        return self._scenes
+
+    @staticmethod
+    def _dims_shape_to_scene_dims_shape(
+        dims_shape: List[Dict], scene_index: int, consistent: bool
+    ) -> Dict[str, Tuple[int, int]]:
+        """
+        This function takes the output of `get_dims_shape()` and returns a
+        dictionary of dimensions for the selected scene
+
+        Parameters
+        ----------
+        dims_shape: List[Dict]
+            a list of dictionaries, generated by `get_dims_shape()`
+        scene_index: int
+            the index of the scene being used
+        consistent: bool
+            true if the dictionaries are consistent could be represented
+            compactly (dims_shape with length 1)
+
+        Returns
+        -------
+        A dictionary of dimensions, ie
+        {"T": (0, 1), "C": (0, 3), "Y": (0, 256), "X":(0, 256)}.
+        """
+        dims_shape_index = 0 if consistent else scene_index
+        dims_shape_dict = dims_shape[dims_shape_index]
+        dims_shape_dict.pop(CZI_SCENE_DIM_CHAR, None)
+        return dims_shape_dict
+
+    @staticmethod
+    def _dim_helper(
+        dim: str,
+        selected: Dict[str, Optional[int]],
+        np_index: Tuple,
+        retrieve_dims: List,
+    ) -> Optional[int]:
+        indices = None
+        if dim in selected:
+            indices = selected[dim]
+            if indices is None:
+                indices = np_index[retrieve_dims.index(dim)]
+        return indices
+
+    @staticmethod
+    def _get_just_image_data(
+        fs: AbstractFileSystem,
+        path: str,
+        scene: int,
+        read_dims: Optional[Dict[str, int]] = None,
+    ) -> np.ndarray:
+        return CziReader._get_image_data(
+            fs=fs, path=path, scene=scene, read_dims=read_dims
+        )[0]
+
+    @staticmethod
+    def _get_image_data(
+        fs: AbstractFileSystem,
+        path: str,
+        scene: int,
+        read_dims: Optional[Dict[str, int]] = None,
+    ) -> Tuple[np.ndarray, List[Tuple[str, int]]]:
+        """
+        Read and return the squeezed image data requested along with the dimension info
+        that was read.
+        Parameters
+        ----------
+        img: Path
+            Path to the CZI file to read.
+        read_dims: Optional[Dict[str, int]]
+            The dimensions to read from the file as a dictionary of string to integer.
+            Default: None (Read all data from the image)
+        Returns
+        -------
+        data: np.ndarray
+            The data read for the dimensions provided.
+        read_dimensions: List[Tuple[str, int]]]
+            The dimension sizes that were returned from the read.
+        """
+        # Catch optional read dim
+
+        if read_dims is None:
+            read_dims = {}
+
+        read_dims["S"] = scene
+
+        # Init czi
+        # resource = fs.open(path)
+        czi = CziFile(path)
```

Comment:
In the case where we do eventually add remote reading, lets use:

```python
with fs.open(path) as open_resource:
    czi = CziFile(open_resouce)
```

if that's not possible to do even right now, then simpy remove the commented out line.